pkg/controller/vitesscluster/reconcile_topo.go

```diff
@@ -85,43 +85,58 @@ func (r *ReconcileVitessCluster) reconcileCellTopology(ctx context.Context, vt *
 		desiredCells[cell.Name] = cell
 	}
 
-	// We need to add an alias for all the cells in each region so that vtgate
-	// knows that it can route traffic between them.
-	// Currently, only one region is supported so we allow routing anywhere.
-	//
-	// We also need to create the aliases before we create the cells because we
-	// don't want any vtgates to start after the cells are created but before
-	// the alias exists.
-	desiredCellsAliases := buildCellsAliases(desiredCells)
-	currentCellsAliases, err := ts.GetCellsAliases(ctx, true)
+	if *vt.Spec.TopologyReconciliation.RegisterCellsAliases {
+		err := r.registerCellsAliases(ctx, vt, ts, desiredCells)
```

Comment:
Bring back the part of the old comment that explained why this has to happen before creating cells, since that explains why we bail out here whereas the other cases simply Merge and continue.

Reply:
The comment was moved into inside of `registerCellsAliases` already. Is that sufficient?

Reply:
I was thinking that part of the comment should stay here because we need to explain why this looks different: here we return immediately (bail out) whereas below we merely Merge. It's harder to make the connection if the comment is buried in a subroutine.

Reply:
Alright, I'll move it then.

Reply:
If this must happen before creating cells, then theoretically we shouldn't registercells unless we also allow registration of cells aliases, correct?

Reply:
Done https://github.com/planetscale/vitess-operator/pull/26/commits/ec1001eb2eee3336b9978ab3bd5a9536a2bcda21.